src/main/java/by/custom/utilcalculator/domain/UserProgress.java

```diff
@@ -5,155 +5,156 @@
 import by.custom.utilcalculator.domain.tree.CommandTree;
 
 import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 public class UserProgress implements Serializable {
-    private TransportType transportType = null;
+    private GeneralTransportType generalTransportType = null;
     private CountryOrigin countryOrigin = null;
     private OwnersType ownersType = null;
     private CarAge carAge = null;
-    private TypeOfEngine typeOfEngine = null;
-    private VolumeOfEngine volumeOfEngine = null;
+    private EngineType engineType = null;
+    private EngineVolumeOrPower engineVolumeOrPower = null;
+    private ParticularTransportType particularTransportType = null;
+    private TruckUnitClass truckUnitClass = null;
+    private Weight weight = null;
+    private TrailerO4Type trailersO4Type = null;
     private final String chatID;
-    private Step currentQuestion;
+    private final List<Command> userPath;
 
     public UserProgress(final String chatID) {
         this.chatID = chatID;
+        userPath = new ArrayList<>();
     }
 
-    public TransportType getTransportType(){
-        return transportType;
+    private void addUserStatusToPath(final StepsIndicator stepsIndicator) {
+        final Map<StepsIndicator, Command> fieldsToCommands = CommandTree.getInstance().getFieldsToCommands();
```

Comment:
Now I think it's better to rely on tree to get step number. Let's discuss it on a call.